### ISSUE TITLE
Duplicated warning

### ISSUE THREAD

**Issue by GoogleCodeExporter** [typetools/checker-framework#373]
```
What steps will reproduce the problem?
1. Run the Null Checker on the class below
import java.util.AbstractMap;
import java.util.Collections;
import java.util.Map;
import java.util.Set;

public class Test extends AbstractMap<String, String> {
    @Override
    public Set<Map.Entry<String, String>> entrySet() {
        return Collections.emptySet();
    }
}

What is the expected output? What do you see instead?
Expected: (ideally) no warning
Actual: The same warning is issued twice
    public Set<Map.Entry<String, String>> entrySet() {
              ^
  found   : Set<Entry<String, String>>
  required: Set<Entry<@KeyFor("this") String, String>>
...
    public Set<Map.Entry<String, String>> entrySet() {
              ^
  found   : Set<Entry<String, String>>
  required: Set<Entry<@KeyFor("this") String, String>>
2 warnings

What version of the product are you using? On what operating system?
1.8.6/Windows 8/Java 8

```

Original issue reported on code.google.com by `ClovisSe...@gmail.com` on 4 Oct 2014 at 11:03
